### PR TITLE
[NN] add egatconv edge_weight

### PR DESCRIPTION
**Pull request by venzino-han**
related to issue #5459

add edge_weight parameter
add test for edge_weight

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by mufeili** [comment]
@dgl-bot

File: tests/python/pytorch/nn/test_nn.py [hunk 1]
```diff
@@ -670,7 +670,7 @@ def test_egat_conv(g, idtype, out_node_feats, out_edge_feats, num_heads):
 
     assert h.shape == (g.num_nodes(), num_heads, out_node_feats)
     assert f.shape == (g.num_edges(), num_heads, out_edge_feats)
-    _, _, attn = egat(g, nfeat, efeat, True)
+    _, _, attn = egat(g, nfeat, efeat, None, True)
```
Comment (mufeili) [inline tests/python/pytorch/nn/test_nn.py:673]:
```suggestion
    _, _, attn = egat(g, nfeat, efeat, get_attention=True)
```

File: tests/python/pytorch/nn/test_nn.py [hunk 2]
```diff
@@ -701,7 +701,35 @@ def test_egat_conv_bi(g, idtype, out_node_feats, out_edge_feats, num_heads):
 
     assert h.shape == (g.number_of_dst_nodes(), num_heads, out_node_feats)
     assert f.shape == (g.num_edges(), num_heads, out_edge_feats)
-    _, _, attn = egat(g, nfeat, efeat, True)
+    _, _, attn = egat(g, nfeat, efeat, None, True)
```
Comment (mufeili) [inline tests/python/pytorch/nn/test_nn.py:704]:
```suggestion
    _, _, attn = egat(g, nfeat, efeat, get_attention=True)
```

**Review by mufeili — APPROVED** [review]
Thank you for the great efforts. I'm good other than some minor comments.

**Comment by venzino-han** [comment]
@mufeili would you please kindly check CI issue?

**Comment by mufeili** [comment]
@dgl-bot

File: tests/python/pytorch/nn/test_nn.py [hunk 3]
```diff
@@ -701,7 +701,35 @@ def test_egat_conv_bi(g, idtype, out_node_feats, out_edge_feats, num_heads):
...
+    egat = nn.EGATConv(
+        in_node_feats=10,
+        in_edge_feats=5,
+        out_node_feats=out_node_feats,
+        out_edge_feats=out_edge_feats,
+        num_heads=num_heads,
+    )
+    nfeat = F.randn((g.num_nodes(), 10))
+    efeat = F.randn((g.num_edges(), 5))
+    ew = F.randn(*(g.num_edges(),))
```
Comment (mufeili) [inline tests/python/pytorch/nn/test_nn.py:727]:
```suggestion
    ew = F.randn((g.num_edges(),))
```

**Comment by venzino-han** [comment]
@mufeili I'll fix this test and let you know.

File: tests/python/pytorch/nn/test_nn.py [hunk 4]
```diff
@@ -701,7 +701,36 @@ def test_egat_conv_bi(g, idtype, out_node_feats, out_edge_feats, num_heads):
...
+    ctx = F.ctx()
+    g = g.astype(idtype).to(ctx)
+    egat = nn.EGATConv(
+        in_node_feats=10,
+        in_edge_feats=5,
+        out_node_feats=out_node_feats,
+        out_edge_feats=out_edge_feats,
+        num_heads=num_heads,
+    )
+    egat = egat.to(ctx)
```
Comment (venzino-han) [inline tests/python/pytorch/nn/test_nn.py:725]: @mufeili I missed this part and this might caused the test error

**Comment by mufeili** [comment]
@dgl-bot

**Pull request merged**